snuba/datasets/factory.py

```diff
@@ -111,7 +113,8 @@ def _ds_factory() -> _DatasetFactory:
     # This function can be acessed by many threads at once. It is okay if more than one thread recreates the same object.
     global _DS_FACTORY
     if _DS_FACTORY is None:
-        _DS_FACTORY = _DatasetFactory()
+        with sentry_sdk.start_span(op="function", description="Load Dataset Factory"):
```

Comment:
this should probably be in the `__init__` function. Where we call this may change and we want the profiling to not dissapear